### DISCUSSION TITLE
How to build and debug in a Linux distribution using VSCode?

### DISCUSSION THREAD

**Discussion by PlatinumLucario** [NickvisionApps/Parabolic#246] [Q&A] [answered]
I must be missing something really important, because attempting to build this results in a System.ArgumentNullException.

![Screenshot_20230521_165557](https://github.com/NickvisionApps/TubeConverter/assets/1906189/f58cba2b-8fee-4fe7-84fb-d5aab38ccd59)

Surely, it must be because it was missing the libraries that Gir.Core needs, right? Well as soon as I put all the needed libraries into the same directory as the executable, this happened:
![Screenshot_20230521_172310](https://github.com/NickvisionApps/TubeConverter/assets/1906189/3c2ef8b9-7488-443b-b8a5-359963590a7a)

It looks like it opened successfully, but nothing opened. So I'm really curious, is there anything I'm missing? I've installed all the dependencies that were listed in the readme.

**Comment by fsobolev** [comment, marked as answer]
> Surely, it must be because it was missing the libraries that Gir.Core needs, right?

No. It's not enough to use only dotnet to build the app, there is extra stuff (gresource and `*ui` files), and so there is `justfile` for a reason. If I'm not mistaken, the error you got initially is because `*.ui` files are not found in the assembly - because they were not compiled using `blueprint-compiler`. I'm not familiar with VSCode enough so I can't say unfortunately how to properly change launch configuration, but what you essentially want is to execute `just run`.
But what happens on the second screenshot is a mystery to me. I can only say that it's not required to move libs to the folder with executable.

**Reply by PlatinumLucario** [reply]
Brilliant! Thank you! I was missing the ``blueprint-compiler`` and ``just`` packages. That's what I needed for how this project is set up.

**Reply by fsobolev** [reply]
My previous answer became obsolete, since we moved to Cake build system and also some parts of building process were moved to `csproj` file, so now `dotnet run` works fine, there's no need for a special command to simply build and run the app (although blueprint-compiler is still needed, but not `just`). But for publishing and installing commands, see CONTRIBUTING.md.
